src/main/scala/zio/redis/options/Strings.scala

```diff
@@ -0,0 +1,39 @@
+package zio.redis.options
+
+trait Strings {
+
+  sealed trait BitFieldType
+  object BitFieldType {
+    sealed case class UnsignedInt(size: Int) extends BitFieldType
+    sealed case class SignedInt(size: Int)   extends BitFieldType
+  }
+
+  case class BitFieldGet(`type`: BitFieldType, offset: Int)
+  case class BitFieldSet(`type`: BitFieldType, offset: Int, value: BigInt)
+  case class BitFieldIncr(`type`: BitFieldType, offset: Int, increment: BigInt)
+
+  sealed trait BitFieldOverflow
+  object BitFieldOverflow {
+    case object FAIL extends BitFieldOverflow
+    case object SAT  extends BitFieldOverflow
+    case object WRAP extends BitFieldOverflow
+  }
+
+  sealed trait BitOperation
+  object BitOperation {
+    case object AND extends BitOperation
+    case object OR  extends BitOperation
+    case object XOR extends BitOperation
+  }
+
+  case class BitPosRange(start: Long, end: Option[Long])
```

Comment:
We can evolve this into general solution, something like:

```scala
sealed trait Range

object Range {
  final case class LeftClosed(a: Long) extends Range
  final case class LeftClosedRightOpen(a: Long, b: Long) extends Range
  final case class Closed(a: Long, b: Long) extends Range
  // ...
}
```

The names can be improved as they're math-inspired, and we probably need only combinations given above. On top of these, we can add some helpers to create them.

Reply:
Below a complete list of all possible combinations between open/closed and bounded/unbounded intervals.

| Interval | Type |
| --------- | ------------- |
| (a,b) | `Open` |
| [a,b] | `Closed` |
| (a,b] | `LeftOpenRightClosed` | 
| [a,b) | `LeftClosedRightOpen` |
| (a,+∞) | `LeftOpenRightUnbounded` |
| [a,+∞) | `LeftClosedRightUnbounded` |
| (−∞,b) | `LeftUnboundedRightOpen` |
| (−∞,b] | `LeftUnboundedRightClosed` |
| (−∞,+∞) | `Unbounded` |

```
sealed trait Range[T]
object Range {
    final case class Open[T](lo:T,hi:T) extends Range[T]
    final case class Closed[T](lo:T,hi:T) extends Range[T]
    final case class LeftOpenRightClosed[T](lo:T,hi:T) extends Range[T]
    final case class LeftClosedRightOpen[T](lo:T,hi:T) extends Range[T]
    final case class LeftOpenRightUnbounded[T](lo:T) extends Range[T]
    final case class LeftClosedRightUnbounded[T](lo:T) extends Range[T]
    final case class LeftUnboundedRightOpen[T](hi:T) extends Range[T]
    final case class LeftUnboundedRightClosed[T](hi:T) extends Range[T]
    final case object Unbounded extends Range[Nothing]
}
```

## Range Usage

Here is an overview of all commands we have made available so far grouped by range types.

### Commands accepting `Closed`

`LTRIM`,`LRANGE`,`ZRANGE`,`ZREVRANGE`,`ZREMRANGEBYRANK`

For each command negative indices means counted from the end of range.

Possible range input type for those commands: `Input[options.Range.Closed[Long]]`.

### Commands accepting `Closed` or `LeftClosedRightUnbounded`

`BITPOS`, `BITCOUNT`

For each command the range is optional.

Possible range input type for those commands:
`Input[Either[options.Range.Closed[Long],options.Range.LeftClosedRightUnbounded[Long]]]`.

### Commands accepting any Interval

`ZCOUNT` (using +inf,-inf for infinite intervals)
`ZRANGEBYSCORE`,`ZREVRANGEBYSCORE`,`ZREMRANGEBYSCORE` (using +inf,-inf for infinite intervals)
`ZLEXCOUNT`,`ZRANGEBYLEX`,`ZREVRANGEBYLEX`,`ZREMRANGEBYLEX` (using +,- for infinite intervals)

Possible range input type for score set commands:
`Input[options.Range[Double]`

Possible range input type for lexicographical set commands:
`Input[options.Range[String]`

Reply:
Nice work! I propose that you open up a separate PR and introduce ranges. In case we don't need some of the options, you can cut down the hierarchy.